spyder/plugins/editor/plugin.py

```diff
@@ -1109,33 +1130,96 @@ def get_plugin_actions(self):
+                                                  ima.icon('redo'))
+            self.copy_action = create_edit_action('Copy', _('Copy'),
+                                                  ima.icon('editcopy'))
+            self.cut_action = create_edit_action('Cut', _('Cut'),
+                                                 ima.icon('editcut'))
+            self.paste_action = create_edit_action('Paste', _('Paste'),
+                                                   ima.icon('editpaste'))
+            self.selectall_action = create_edit_action("Select All",
+                                                       _("Select All"),
+                                                       ima.icon('selectall'))
+
+            for action in [self.undo_action, self.redo_action]:
+                mainmenu.add_item_to_application_menu(
+                    action,
+                    menu_id=ApplicationMenus.Edit,
+                    section=EditMenuSections.UndoRedo,
+                    before_section=EditMenuSections.Editor,
+                    omit_id=True
+                )
 
-        self.main.search_menu_actions = (
-            search_menu_actions + self.main.search_menu_actions)
+            for action in [
+                    self.cut_action, self.copy_action, self.paste_action,
+                    self.selectall_action
+                    ]:
+                mainmenu.add_item_to_application_menu(
+                    action,
+                    menu_id=ApplicationMenus.Edit,
+                    section=EditMenuSections.Copy,
+                    before_section=EditMenuSections.Editor,
+                    omit_id=True
+                )
 
-        # ---- Source menu/toolbar construction ----
-        source_menu_actions = [
+            for search_item in self.edit_menu_actions:
+                mainmenu.add_item_to_application_menu(
+                    search_item,
+                    omit_id=True,
+                    menu_id=ApplicationMenus.Edit,
+                    before_section=SearchMenuSections.FindInFiles,
+                    section=EditMenuSections.Editor
+                )
+
+        # ---- Search menu construction ----
+        search_menu_actions = [self.find_action,
+                               self.find_next_action,
+                               self.find_previous_action,
+                               self.replace_action,
+                               self.gotoline_action]
+        if mainmenu:
+            search_menu = mainmenu.get_application_menu(ApplicationMenus.Search)
+            search_menu.aboutToShow.connect(self.update_search_menu)
+            for search_item in search_menu_actions:
+                mainmenu.add_item_to_application_menu(
+                    search_item,
+                    omit_id=True,
+                    menu_id=ApplicationMenus.Search,
+                    before_section=SearchMenuSections.FindInFiles,
+                    section=SearchMenuSections.FindInText
```

Comment:
```suggestion
                    section=SearchMenuSections.FindInText,
                    before_section=SearchMenuSections.FindInFiles
```

I think this a bit clearer to read.